test/compiler.test.js

```diff
@@ -1195,6 +1324,47 @@ describe('compiler', function() {
         }
       });
     });
+
+    it('loads component relative to appRootDir', function() {
+      appdir.writeConfigFileSync('./component-config.json', {
+        './component': { }
+      });
+      var appJS = appdir.writeConfigFileSync('component.js', '');
+
+      var instructions = boot.compile(appdir.PATH);
+      expect(instructions.components[0]).have.property(
+        'sourceFile', appJS
+      );
+    });
+
+    it('loads component relative to node modules', function() {
+      appdir.writeConfigFileSync('component-config.json', {
+        'component': { }
+      });
+      var appJS = appdir.writeConfigFileSync('node_modules/component.js', '');
+
+      var instructions = boot.compile(appdir.PATH);
+      expect(instructions.components[0]).have.property(
+        'sourceFile', appJS
+      );
+    });
+
+    it('prefers coffeescript over json when loading component', function() {
+      require('coffee-script/register');
+      appdir.writeConfigFileSync('component-config.json', {
+        './component': { }
+      });
+
+      var coffee = appdir.writeFileSync('component.coffee', '');
+      appdir.writeFileSync('component.json', '');
+
+      var instructions = boot.compile(appdir.PATH);
+
+      expect(instructions.components).to.have.length(1);
+      expect(instructions.components[0]).have.property(
+        'sourceFile', coffee);
+    });
```

Comment:
Please another test to verify that `node_modules/component/component.coffee` is correctly picked too when the component-config json contains `component` key.